parser/src/format/csv.rs

```diff
@@ -234,12 +234,17 @@ struct Header {
 
 impl Header {
     fn parse(&self, value: &str) -> Result<Value, Error> {
+        // If we don't know any actual type information about this field, then always treat it as a
+        // string.
+        if self.projection.possible_types == types::INVALID {
```

Comment:
Crap, good point. I went back and forth about whether to wrap in an option or not, but I think it's probably worth doing.

Reply:
Is there a meaningful difference between adding another variant to the enum vs wrapping it with an Option? Should we name the `None` concept you describe more explicitly?

Reply:
The enum is type information drawn from static inference of the schema, so in this context IMO Option makes more sense, because semantically it's a distinction between "I know this location has this typeinfo" vs "I have no typeinfo at all".

Put another way, it _wouldn't_ be reasonable for inference to ever return a None equivalent from a schema.